File: Documentation/5.4/Samples/python/Indexes/WhatAreIndexes.py
from ravendb import AbstractIndexCreationTask

from examples_base import ExampleBase, Employee


# region indexes_1
class Employees_ByFirstAndLastName(AbstractIndexCreationTask):
    def __init__(self):
        super().__init__()
        self.map = "from employee in docs.Employees select new {first_name = employee.FirstName, last_name = employee.LastName}"


# endregion


class WhatAreIndexes(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_what_are_indexes(self):
        with self.embedded_server.get_document_store("WhatAreIndexes") as store:
            with store.open_session() as session:
                # region indexes_2
                # save index on server
                Employees_ByFirstAndLastName().execute(store)
                # endregion

                with store.open_session() as session:
                    # region indexes_3
                    results = list(
                        session.query_index_type(Employees_ByFirstAndLastName, Employee).where_equals(
                            "first_name", "Robert"
                        )
                    )
                    # endregion

File: Documentation/5.4/Samples/php/Indexes/WhatAreIndexes.php
<?php

namespace RavenDB\Samples\Indexes;

use RavenDB\Documents\DocumentStore;
use RavenDB\Documents\Indexes\AbstractIndexCreationTask;
use RavenDB\Samples\Infrastructure\Orders\Employee;

# region indexes_1
class Employees_ByFirstAndLastName extends AbstractIndexCreationTask
{
    public function __construct()
    {
        parent::__construct();

        $this->map = "docs.Employees.Select(employee => new {" .
            "    FirstName = employee.FirstName," .
            "    LastName = employee.LastName" .
            "})";
    }
}
# endregion


class WhatAreIndexes
{
    public function samples(): void
    {
        $store = new DocumentStore();

        try {
            # region indexes_2
            // save index on server
            (new Employees_ByFirstAndLastName())->execute($store);
            # endregion

            $session = $store->openSession();
            try {
                # region indexes_3
                /** @var array<Employee> $results */
                $results = $session->query(Employee::class, Employees_ByFirstAndLastName::class)
                        ->whereEquals('FirstName', "Robert")
                        ->toList();
                # endregion
            } finally {
                $session->close();
            }
        } finally {
            $store->close();
        }
    }
}
